merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/services/TypescriptCodeGenerationService.java

```diff
@@ -33,4 +166,98 @@ private static String joinLines(final Iterable<String> result) {
   private static String indent(final String s) {
     return joinLines(s.lines().map(line -> "  " + line).toList());
   }
+
+
+  private static String valueSchemaToTypescript(final ValueSchema valueSchema) {
+    return valueSchema.match(new ValueSchema.Visitor<>() {
+      @Override
+      public String onReal() {
+        return "number";
+      }
+
+      @Override
+      public String onInt() {
+        return "number";
+      }
+
+      @Override
+      public String onBoolean() {
+        return "boolean";
+      }
+
+      @Override
+      public String onString() {
+        return "string";
+      }
+
+      @Override
+      public String onDuration() {
+        return "Duration";
+      }
+
+      @Override
+      public String onPath() {
+        return "string";
+      }
+
+      @Override
+      public String onSeries(final ValueSchema value) {
+        return valueSchemaToTypescript(value) + "[]";
+      }
+
+      @Override
+      public String onStruct(final Map<String, ValueSchema> values) {
+        final var result = new StringBuilder("{");
+
+        for (final var member: values.keySet()) {
+          result
+              .append(member)
+              .append(": ")
+              .append(valueSchemaToTypescript(values.get(member)))
+              .append(", ");
+        }
+
+        result.append("}");
+        return result.toString();
+      }
+
+      @Override
+      public String onVariant(final List<ValueSchema.Variant> variants) {
+        final var result = new StringBuilder("(");
+
+        for (final var variant: variants) {
+          result
+              .append(" | \"")
+              .append(variant.label())
+              .append("\"");
+        }
+
+        result.append(")");
+        return result.toString();
+      }
+    });
+  }
+
+  private static String valueSchemaToTypescriptProfile(final ValueSchema valueSchema) {
+    var basicTypescript = valueSchemaToTypescript(valueSchema);
+    if (basicTypescript.equals("number")) {
+      return "Real";
+    } else {
+      return "Discrete<" + basicTypescript + ">";
+    }
+  }
+
+  private static boolean valueSchemaIsReal(final ValueSchema valueSchema) {
+    return valueSchema.match(new ValueSchema.DefaultVisitor<>() {
+      @Override
+      protected Boolean onDefault() {
+        return false;
+      }
+
```

Comment:
Any objections to making this `valueSchemaIsNumeric`, and adding an `onInt() { return true; }` case? I can't think of a reason that would bite us - it should be okay to use integers for Real profiles 🤔 

Context: for the scheduler tests, I've been using the `/plant` resource for all of my numeric comparisons. I ran into this error:
```
Argument of type '"/plant"' is not assignable to parameter of type 'RealResourceName'
```

Reply:
done

Reply:
I have one more case I'd like to add to `valueSchemaIsNumeric` (I don't need to hold up this PR, I can add it when I'm rebasing as well 👍)

I finally understood my confusion about `Real` vs `Linear` - `Linear` resources are serialized as `{initial: number, rate: number}`, and the constraints backend will handle all the interpolation. On the frontend, I think we can treat them as numeric. Here's my proposal for how `valueSchemaIsNumeric` would look:

```java
private static boolean valueSchemaIsNumeric(final ValueSchema valueSchema) {
    return valueSchema.equals(ValueSchema.INT)
        || valueSchema.equals(ValueSchema.REAL)
        || valueSchema.equals(LINEAR_RESOURCE_SCHEMA);
  }
```

where `LINEAR_RESOURCE_SCHEMA` is defined as so:

```java
private static final ValueSchema LINEAR_RESOURCE_SCHEMA = ValueSchema.ofStruct(Map.of(
      "initial", ValueSchema.REAL,
      "rate", ValueSchema.REAL));
```

I've changed `/fruit` to be a linear resource in my branch and added a test with this scheduling goal to the scheduler integration tests:
```typescript
export default (): Goal => {
 return Goal.CoexistenceGoal({
   activityTemplate: ActivityTemplates.PeelBanana({peelDirection: "fromStem"}),
   forEach: Windows.All(
     Real.Resource("/fruit").lessThan(4.0),
     Real.Resource("/fruit").greaterThan(2.0)
   )
 })
```